versions/development/SPEC.md

```diff
@@ -2705,12 +2705,42 @@ In this example, `i`, and `f` are inputs to this task even though `i` is not dir
 
 ## Computing Workflow Inputs
 
-Workflows have inputs that must be satisfied to run them, just like tasks. Inputs to the workflow are provided as a key/value map where the key is of the form `workflow_name.input_name`.
+Workflows have inputs that must be satisfied to run them, just like tasks. 
+Inputs to the workflow are provided as a key/value map where the key is of the 
+form `workflow_name.input_name`.
+
+* A call has its inputs supplied when called by a workflow. 
+    * Example: `call my_task { input: my_task_input=... }`
+* All required call inputs which do not have defaults should be filled by the 
+  calling workflow.
+* A workflow is allowed not to specify optional inputs in a call's input block. 
+  In this case, the inputs bubble up to become a nested input to the workflow 
+  instead.
+    * Example: an unsupplied input might have the fully-qualified name 
+      `my_workflow.my_task.my_task_input`.
+* If that workflow is used as a subworkflow, the input is allowed to bubble up 
+  again with a further-qualified name.
+    * Example: my_outer_workflow.my_workflow.my_task.my_task_input.
+* There is currently no way to supply a bubbled-up input in an outer workflow's 
+  call block.
+    * Example: the following inputs would not be valid for a call to a subworkflow
+      `{ inputs: my_task.my_task_input=... }`
+* By default an engine only allows inputs that are specified in the input
+  section of the top-level workflow.
+* If a workflow is suitable for use with nested inputs it should be explicitly
+  stated. This is done by setting `allowNestedInputs` to `true` in the meta
```

Comment:
My problem with this is that things like `allowNestedInputs` don't really belong in `meta` - there's just no where else to put them. I am going to open a proposal to add a `hints` section to workflow with the same semantics as it has in task. I think things like this more appropriately belong in hints than in meta.

Reply:
I think  when designing the `hints` section, I ultimately planned on having one in the `workflow` as well. however it felt like a more natural transition to start with it in the task. In the lont term this is probably a good move. If we add the hints in 1.1, I feel like 2.0 can propogate the hints to the workflow

Reply:
I agree! Meta was always a bit of a "let's put it where it doesn't hurt" manoeuver. 

Not having this feature proved to be a major hindrance in getting actual work done. But at least it is implemented in cromwell now, so it is at least possible to get work done until this feature lands.

If the hints proposal is landed I will rewrite this PR.